Quarkus 2.16.6: MQTT Connector allways starts dev services although all config is provided

### Describe the bug

Since Quarkus 2.16.6, you cannot start quarkusDev without having a running docker when you use the MQTT connector. Even though I configured everything required to connect to my external MQTT provider, Quarkus tries to start the MQTT dev services and crashes when you don't have docker running. 

My config (also see reproducer below) to a public test broker:

```
mp.messaging.incoming.consumer1.connector=smallrye-mqtt
mp.messaging.incoming.consumer1.host=public.mqtthq.com
mp.messaging.incoming.consumer1.port=1883
mp.messaging.incoming.consumer1.topic=arconsis/#
```

With Quarkus 2.16.5.Final, the application did start and not crash. 

A workaround is to disable devservices with this config:

```
quarkus.devservices.enabled=false
```

### Expected behavior

Quarkus should not start the dev services because I configured the required properties.

### Actual behavior

At startup, Quarkus crashes if you do not have docker running because it tries to start dev services. This is the exception:

```
2023-04-06 11:48:29,298 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-1) Loaded org.testcontainers.dockerclient.NpipeSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2023-04-06 11:48:29,326 INFO  [org.tes.doc.DockerMachineClientProviderStrategy] (build-1) docker-machine executable was not found on PATH ([ ... ])
2023-04-06 11:48:29,327 ERROR [org.tes.doc.DockerClientProviderStrategy] (build-1) Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
As no valid configuration was found, execution cannot continue.
See https://www.testcontainers.org/on_failure.html for more details.
2023-04-06 11:48:29,531 INFO  [org.tes.uti.ImageNameSubstitutor] (build-1) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2023-04-06 11:48:29,550 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2023-04-06 11:48:29,608 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor#startMqttDevService threw an exception: java.lang.RuntimeException: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:98)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:212)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:150)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:186)
	at org.testcontainers.DockerClientFactory$1.getDockerClient(DockerClientFactory.java:104)
	at com.github.dockerjava.api.DockerClientDelegate.listContainersCmd(DockerClientDelegate.java:188)
	at io.quarkus.devservices.common.ContainerLocator.lookup(ContainerLocator.java:32)
	at io.quarkus.devservices.common.ContainerLocator.locateContainer(ContainerLocator.java:45)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttBroker(MqttDevServicesProcessor.java:175)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:81)
	... 11 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:335)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:86)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
```

### How to Reproduce?

Reproducer: 
[2023-04-06-mqtt-always-devservices.zip](https://github.com/quarkusio/quarkus/files/11168943/2023-04-06-mqtt-always-devservices.zip)

1. Download and unzip the reproducer
2. Stop your local docker
3. start the reproducer with `./gradlew quarkusDev`
4. See the crash in the console

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

JDK17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.16.6.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

Hi @geoand, 

I just tried it with Quarkus 3.0.0.CR2 and it shows the same behavior. I used the reproducer and just updated it to the new Quarkus version. Here are the logs:

```
2023-04-12 09:47:41,349 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-4) Loaded org.testcontainers.dockerclient.NpipeSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2023-04-12 09:47:41,376 INFO  [org.tes.doc.DockerMachineClientProviderStrategy] (build-4) docker-machine executable was not found on PATH ([ ... ])
2023-04-12 09:47:41,380 ERROR [org.tes.doc.DockerClientProviderStrategy] (build-4) Could not find a valid Docker environment. Please check configuration. Attempted configurations were:
As no valid configuration was found, execution cannot continue.
See https://www.testcontainers.org/on_failure.html for more details.
2023-04-12 09:47:41,809 INFO  [org.tes.uti.ImageNameSubstitutor] (build-4) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2023-04-12 09:47:41,841 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2023-04-12 09:47:41,902 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor#startMqttDevService threw an exception: java.lang.RuntimeException: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:98)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:212)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:150)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:186)
	at org.testcontainers.DockerClientFactory$1.getDockerClient(DockerClientFactory.java:104)
	at com.github.dockerjava.api.DockerClientDelegate.listContainersCmd(DockerClientDelegate.java:188)
	at io.quarkus.devservices.common.ContainerLocator.lookup(ContainerLocator.java:32)
	at io.quarkus.devservices.common.ContainerLocator.locateContainer(ContainerLocator.java:45)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttBroker(MqttDevServicesProcessor.java:175)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:81)
	... 11 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:335)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:86)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor#startMqttDevService threw an exception: java.lang.RuntimeException: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:98)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:212)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:150)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:186)
	at org.testcontainers.DockerClientFactory$1.getDockerClient(DockerClientFactory.java:104)
	at com.github.dockerjava.api.DockerClientDelegate.listContainersCmd(DockerClientDelegate.java:188)
	at io.quarkus.devservices.common.ContainerLocator.lookup(ContainerLocator.java:32)
	at io.quarkus.devservices.common.ContainerLocator.locateContainer(ContainerLocator.java:45)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttBroker(MqttDevServicesProcessor.java:175)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:81)
	... 11 more

	at io.quarkus.builder.Execution.run(Execution.java:123)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:331)
	... 9 more
Caused by: java.lang.RuntimeException: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:98)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:212)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:150)
	at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:186)
	at org.testcontainers.DockerClientFactory$1.getDockerClient(DockerClientFactory.java:104)
	at com.github.dockerjava.api.DockerClientDelegate.listContainersCmd(DockerClientDelegate.java:188)
	at io.quarkus.devservices.common.ContainerLocator.lookup(ContainerLocator.java:32)
	at io.quarkus.devservices.common.ContainerLocator.locateContainer(ContainerLocator.java:45)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttBroker(MqttDevServicesProcessor.java:175)
	at io.quarkus.smallrye.reactivemessaging.mqtt.deployment.MqttDevServicesProcessor.startMqttDevService(MqttDevServicesProcessor.java:81)
	... 11 more

```

--- a/extensions/smallrye-reactive-messaging-mqtt/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/mqtt/deployment/MqttDevServicesProcessor.java
+++ b/extensions/smallrye-reactive-messaging-mqtt/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/mqtt/deployment/MqttDevServicesProcessor.java
@@ -203,17 +203,16 @@ public class MqttDevServicesProcessor {
             boolean isIncoming = name.startsWith("mp.messaging.incoming.");
             boolean isOutgoing = name.startsWith("mp.messaging.outgoing.");
             boolean isConnector = name.endsWith(".connector");
-            boolean isConfigured = false;
+            boolean isConfigured;
             if ((isIncoming || isOutgoing) && isConnector) {
                 String connectorValue = config.getValue(name, String.class);
                 boolean isMqtt = connectorValue.equalsIgnoreCase("smallrye-mqtt");
                 boolean hasHost = ConfigUtils.isPropertyPresent(name.replace(".connector", ".host"));
                 boolean hasPort = ConfigUtils.isPropertyPresent(name.replace(".connector", ".port"));
                 isConfigured = isMqtt && (hasHost || hasPort);
-            }
-
-            if (!isConfigured) {
-                return true;
+                if (!isConfigured) {
+                    return true;
+                }
             }
         }
         return false;

